model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -184,55 +187,72 @@ message ProcessBundleDescriptor {
   org.apache.beam.model.pipeline.v1.ApiServiceDescriptor state_api_service_descriptor = 7;
 }
 
-// Represents a partition of the bundle into two bundles: a "primary" and
-// a "residual", with the following properties:
-// - The work in primary and residual doesn't overlap, and combined, adds up
-//   to the work in the current bundle if the split hadn't happened.
-// - The current bundle, if it keeps executing, will have done none of the
-//   work under residual roots.
-// - The current bundle, if no further splits happen, will have done exactly
-//   the work under primary_roots.
-// For more rigorous definitions see https://s.apache.org/beam-breaking-fusion
-message BundleSplit {
-  // One of the root applications specifying the scope of work for a bundle.
-  message Application {
-    // (Required) The primitive transform to which to pass the element
-    string ptransform_id = 1;
+// One of the applications specifying the scope of work for a bundle.
+message BundleApplication {
+  // (Required) The primitive transform to which to pass the element
+  string ptransform_id = 1;
+
+  // (Required) Name of the transform's input to which to pass the element.
+  string input_id = 2;
 
-    // (Required) Name of the transform's input to which to pass the element.
-    string input_id = 2;
+  // (Required) The encoded element to pass to the transform.
+  bytes element = 3;
 
-    // (Required) The encoded element to pass to the transform.
-    bytes element = 3;
+  // Lower bound on timestamps of elements that this PTransform
```

Comment:
I find this description somewhat hard to grok. Some thoughts:

1. Start with what the map is keyed by (rather than ending with it), since this context is useful for the rest of the explanation.
2. Is the explanation here trying to define what an output watermark is? I find the explanation makes it harder to understand the usage of this particular field. Maybe it'd be simpler to just refer to it as an output watermark and let it be defined more fully somewhere else.

Reply:
Reworded, thanks.